transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -343,4 +351,50 @@ public void close() {
             channel.close();
         }
     }
+
+    @Override
+    public Stats stats() {
+        return stats;
+    }
+
+    private class ChannelPoolStats implements ChannelPool.Stats, ChannelPoolHandler {
+        private final ChannelPoolHandler delegate;
+        private final AtomicInteger leasedChannelCount = new AtomicInteger(0);
+
+        public ChannelPoolStats(ChannelPoolHandler delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public void channelCreated(Channel ch) throws Exception {
+            delegate.channelCreated(ch);
+        }
+
+        @Override
+        public void channelReleased(Channel ch) throws Exception {
+            leasedChannelCount.decrementAndGet();
+            delegate.channelReleased(ch);
+        }
+
+        @Override
+        public void channelAcquired(Channel ch) throws Exception {
+            leasedChannelCount.incrementAndGet();
+            delegate.channelAcquired(ch);
+        }
+
+        @Override
+        public int leasedChannelsCount() {
+            return leasedChannelCount.get();
+        }
+
+        @Override
+        public int availableChannelsCount() {
+            return deque.size();
```

Comment:
Yeah, the size() method can report inaccurate value. 
Two options
- We can document that method to notify clients about the inaccuracy
- Track ourselves the available count